adapters/handlers/rest/handlers_authz.go

```diff
@@ -59,20 +59,43 @@ func (h *authZHandlers) createRole(params authz.CreateRoleParams, principal *mod
 		return authz.NewCreateRoleUnprocessableEntity().WithPayload(errPayloadFromSingleErr(errors.New("role name is required")))
 	}
 
-	err := h.controller.CreateRoles(params.Body)
+	err := h.controller.UpsertRoles(params.Body)
 	if err != nil {
 		return authz.NewCreateRoleInternalServerError().WithPayload(errPayloadFromSingleErr(err))
 	}
 
 	return authz.NewCreateRoleCreated()
 }
 
-func (h *authZHandlers) addPermission(params authz.AddPermissionParams, principal *models.Principal) middleware.Responder {
-	panic("not implemented")
+func (h *authZHandlers) addPermissions(params authz.AddPermissionsParams, principal *models.Principal) middleware.Responder {
+	// TODO validate and audit log
+	if err := h.authorizer.Authorize(principal, authorization.UPDATE, authorization.Roles()...); err != nil {
+		return authz.NewCreateRoleInternalServerError().WithPayload(errPayloadFromSingleErr(err))
+	}
+
+	err := h.controller.UpsertRoles(&models.Role{
+		Name:        params.Body.Name,
```

Comment:
**todo (if-minor):**  it needs validation but could be pushed later on another PR

Reply:
This sort of validation is handled by the API framework itself, I believe. So sending a `POST /authz/roles/add-permissions` without `name` in the body will automatically throw a `422` error

Agreed that we should put some validation into `UpsertRoles` just in case, especially if such a method ends up being used elsewhere